parser/grammar.go

```diff
@@ -53,6 +53,106 @@ func (b *Bool) Capture(values []string) error {
 	return nil
 }
 
+type Block struct {
+	Comments []*Comment      `@Comment*`
+	Body     []*BlockElement `(@@ ";")*`
+}
+
+type BlockElement struct {
+	Check     *Check         `@@`
+	Predicate *Predicate     `|@@`
+	RuleBody  []*RuleElement `("<-" @@ ("," @@)*)?`
+}
+
+func (b *Block) ToBiscuit() (*biscuit.ParsedBlock, error) {
+	facts := []biscuit.Fact{}
+	rules := []biscuit.Rule{}
+	checks := []biscuit.Check{}
+	for _, e := range b.Body {
+		if e.Check != nil {
+			c, err := e.Check.ToBiscuit()
+			if err != nil {
+				return nil, err
+			}
+			checks = append(checks, *c)
+		} else if e.Predicate != nil && e.RuleBody != nil {
+			rule := Rule{
+				Head: e.Predicate,
+				Body: e.RuleBody,
+			}
+			r, err := rule.ToBiscuit()
+			if err != nil {
+				return nil, err
+			}
+			rules = append(rules, *r)
+		} else {
```

Comment:
would be really nice to have some exhaustivity check here instead of this (although I am not aware of simple solutions to have it)

Reply:
yeah, go makes this kind of things more difficult than needed :-/